# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Посмотрим на уникальные значения переменных `children`(наличие детей у клиентов) и `debt`(наличие задолженности по кредиту)

In [22]:
data['children'].value_counts()# уникальные значения в столбце children

0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: children, dtype: int64

In [23]:
data['debt'].value_counts()# уникальные значения в столбце debt

0    19599
1     1732
Name: debt, dtype: int64

Видно, что большинство заемщиков не имеют задолженности и не имеет детей.Сгруппируем данные по столбцам`children` и `debt`
рассчитаем встроенными функциями `count()`, `sum()`,`mean()` соответственно значения общего количества заемщиков,количество клиентов, имевших задолженность по возврату кредитов (debt = True -- задолженность была)  и долю клиентов,имеющих задолженность.
Выведем их в отдельную таблицу.


In [24]:

child_of_debt = data.groupby('children')['debt'].agg({'count', 'sum', 'mean'}) 
child_of_debt

count   sum      mean
children                       
0         14091  1063  0.075438
1          4808   444  0.092346
2          2052   194  0.094542
3           330    27  0.081818
4            41     4  0.097561
5             9     0  0.000000

Видно,что клиенты без детей чаще отдают кредит в срок(**7.5%** из них не платят вовремя).В то время, как клиенты с детьми чаще не оплачивают вовремя(процент в этом случае от **8.2 до 9.4%**).Но от количества детей этот показатель,как мы видим напрямую не зависит.Сгруппируем клиентов банка по признаку:есть дети/нет детей и посмотрим на картину вцелом.

In [25]:
def s_children(children): # функция для группировки
    if children > 0: return 'есть дети'
    else: return 'нет детей'
data['child_ex'] = data['children'].apply(s_children)


In [26]:
child_of_debt_new = data.groupby('child_ex')['debt'].agg({'count','sum','mean'})

child_of_debt_new

count   sum      mean
child_ex                        
есть дети   7240   669  0.092403
нет детей  14091  1063  0.075438

**Вывод:** 
На основании проведенного анализа можно сделать следующие выводы:
 - Небольшая зависимость между наличием детей и возвратом кредита есть. Заемщик без детей платит по кредиту в срок чаще, чем заемщик с детьми.(**7.5% против 9.2%**).Возможно на это влияет факт наличия "свободных денег", ведь когда есть дети в семье,то возможны незапланированные расходы.
 - Но среди заемщиков с детьми уже сложнее понять зависимость - можно было бы предположить, что чем больше детей, тем чаще заемщик не выплачивает кредит в срок, но процент должников при 1,2,3,4 и 5 детях не дает нам подтвердить гипотезу о зависимости. 

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Посмотрим на уникальные знчения признака `family_status`

In [27]:
data['family_status'].value_counts() # уникальные значения семейного статуса

женат / замужем          12261
гражданский брак          4134
Не женат / не замужем     2796
в разводе                 1189
вдовец / вдова             951
Name: family_status, dtype: int64

Сгруппируем данные по столбцам`family_status` и `debt` и посмотрим  долю клиентов,имеющих задолженность в зависимости от семейного статуса.
Выведем такхе их в отдельную таблицу.

In [28]:
data_group_fam_stat = data.groupby('family_status').agg({'debt': 'mean'})
data_group_fam_stat.sort_values('debt', ascending = False)

debt
family_status                  
Не женат / не замужем  0.097639
гражданский брак       0.093130
женат / замужем        0.075606
в разводе              0.070648
вдовец / вдова         0.066246

**Вывод:** 
- Из полученных данных видно, что процент невозврата кредита вовремя для групп такой:
  не женат / не замужем = 9,8%
  гражданский брак = 9,3%
  женат / замужем = 7,5%
  в разводе = 7,1%
  вдовец / вдова = 6,6%
- Можно педположить,что клиентов состоящих в браке('женат / замужем') или бывшие ранее в официальном браке('вдовец / вдова' или   'в разводе') можно назвать боллее надежными.
- И напротив клиенты имеющие статус 'Не женат / не замужем' и 'гражданский брак' менее надежны.
- Таким образом можно сказать, что зависимость между семейным положением и возвратом кредита в срок есть.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Выше была проведена категоризация переменной `total_income`(уровень дохода),был создан столбец `total_income_category` с категориями A,B,C,D,E.
- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.
прсмотрим на минимальный,максимальнвый доход клиетов банка и как они распределились по категориям

In [29]:
data['total_income'].min() # минимальный доход

20667

In [30]:
data['total_income'].max() # максимальный доход

2265604

In [31]:
data['total_income_category'].value_counts() # выводим данные из столбца "total_income_category"

C    15921
B     5014
D      349
A       25
E       22
Name: total_income_category, dtype: int64

Видно, что минимальный и максимальный доход отличается практически в  100 раз! И данные по группам распределились неравномерно.
Стоит видимо провести категоризацию признака другим способом.Сложность есть в определении границ групп.В этом случае можно воспользоваться методом `.gcut()` и разделить ежемесячный доход на 5 групп. Названия этим группам дадим следующие: `очень низкий доход`,  `низкий доход`, `средний доход`, `доход выше среднего`, `очень высокий доход`. Запишем результат в исходную таблицу в новую переменную `cat_total_income`

<div class="alert alert-block alert-success">
<b>✔️ Success:</b> все верно! qcut идеально подходит для такой задачи

In [32]:
data['cat_total_income'] = pd.qcut(data['total_income'], 5, labels=['очень низкий доход', 'низкий доход', 'средний доход', 'доход выше среднего', 'очень высокий доход'])
data['cat_total_income']

0        очень высокий доход
1               низкий доход
2              средний доход
3        очень высокий доход
4              средний доход
                ...         
21520    очень высокий доход
21521          средний доход
21522     очень низкий доход
21523    очень высокий доход
21524     очень низкий доход
Name: cat_total_income, Length: 21331, dtype: category
Categories (5, object): ['очень низкий доход' < 'низкий доход' < 'средний доход' < 'доход выше среднего' < 'очень высокий доход']

In [33]:
data.pivot_table(index=['cat_total_income'], values='debt', aggfunc=['sum','count','mean']) # сводная таблица мнтодом pivot_table

sum count      mean
                    debt  debt      debt
cat_total_income                        
очень низкий доход   344  4267  0.080619
низкий доход         358  4266  0.083919
средний доход        373  4266  0.087436
доход выше среднего  358  4266  0.083919
очень высокий доход  299  4266  0.070089

**Вывод:** 

- Из полученных результатов видно, что процент невозврата кредита вовремя в зависимости от уровня дохода такой:
  очень низкий доход= 8,0%
  низкий доход = 8,3%
  средний доход = 8,7%
  доход выше среднего = 8,3%
  очень высокий доход = 7,0%
- Видно, что люди с высоким доходом более ответственно относятся к соблюдению сроков погашения кредита.(У них видимо проблем с     возвратом меньше).
- Наименее надежны клиенты со средним уровнем дохода.
- Таким образом можно сказать, что зависимость между уровнем дохода и возвратом кредита в срок имеетя.Но однозначно на мой         взгляд к ней не стоит относится.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Ранее при категоризации данных(задание 18) из столбца `purpose`(цели кредита) был сформирован  новый столбец `purpose_category`   с категориями.Посмотрим на их значения.

In [34]:
data['purpose_category'].value_counts()  

операции с недвижимостью    10751
операции с автомобилем       4279
получение образования        3988
проведение свадьбы           2313
Name: purpose_category, dtype: int64

Сформируем сводную таблицу

In [35]:
data.pivot_table(index=['purpose_category'], values='debt', aggfunc=['sum','count','mean'])

sum  count      mean
                         debt   debt      debt
purpose_category                              
операции с автомобилем    400   4279  0.093480
операции с недвижимостью  780  10751  0.072551
получение образования     369   3988  0.092528
проведение свадьбы        183   2313  0.079118

**Вывод:** 

Из полученных результатов видно:
 - по операциям с недвижимостью процент невозврата кредита вовремя самый низкий(**7.2%**) и число таких заемщиков больше всего.
   (видимо люди, берущие кредит на эти цели более ответственны)
 - самый высокий процент невозврата кредита вовремя по операциям, связанным с приобретением автомобиля и получением                образования(**9.3% и 9.2% соответственно**).
 - Можно заметить, что цель кредита влияет на срок его возврата.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 

Причинами, приводящими к появлению пропусков в данных , являются:
 - невозможность получения данных или проблемы обработки(произошел технический сбой: поломка аппаратуры снимающей,например,показания температуры объекта)
 - искажение информации(ошибки,связанные с человеческим фактором при ручном введении данных:забыли что-то указать или ввели с ошибкой)
 - сокрытие информации(клиент намеренно не указал сведения о себе или данные имеют скрытый характер по соображениям безопасности).

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 

Для колличественных переменных в случае,если пропусков мало - самое простое,что можно сделать - зто удалить строки с пропущенными значениями.Однако если пропусков достаточно много,рекрмендуется замена на среднее значение,медиану и т.п.(простые способы) Заполнить пропуски медианным значением — лучшее решение, так как медиана – значение, которое делит отсортированные по возрастанию данные на две равные части и в отличии от среднего значения, на нее не влияют выбросы(нулевые и крайне высокие значения)

### Шаг 4: общий вывод.

***В результате выполнения проекта были решены следуюшие задачи:***

**Шаг 1:** 
  Выполненна загрузка данных и предварительное знакомство с ними.Таблица представляет собой данные о клиентах банка и их           платежеспособности.На этом этапе обнаружено:
  - Заголовки  столбцов корректны и исправления не требуют.
  - Есть пропуски в признаках `days_employed` и `total_income`
  - В `days_employed`(количество дней трудового стажа) есть отрицательные значения, что является аномалией
  - В `total_income` тип данных вещественный, удобнее его будет заменить на целочисленный.

**Шаг 2:** 
  Выполнена предобрабработка данных:
  - Проведена замена пропусковв столбцах `days_employed` и `total_income` на медианные значения
  - В столбце `days_employed` всеотрицательные значения заменены на положительные с помощью метода `abs()`
  - Удалены строки с аномальным значением признака `children`(**-1 и 20**)
  - Обработаны неявные дубликаты в столбце `education`
  - Проведена категоризация  для признаков `total_income` и `purpose`, что помогло в дальнейших исследованиях
  
 
**Шаг 3:** 
     **Исследовательский анализ данных:**
     
   **1.Есть ли зависимость между количеством детей и возвратом кредита в срок?**
    
 
 - Выявлена зависимость между наличием детей и возвратом кредита. Заемщик без детей платит по кредиту в срок чаще, чем заемщик      с детьми.(**7.5% против 9.2%**).Но количество детей напрямую не влияет на платежеспособность.
  
  **2.Есть ли зависимость между семейным положением и возвратом кредита в срок?** 
    
  
  - Зависимость между семейным положением и возвратом кредита в срок есть.Клиентов состоящих вбраке('женат / замужем') или           бывшие ранее в официальном браке('вдовец / вдова' или   'в разводе') можно назвать боллее надежными.И напротив,клиенты           имеющие статус 'Не женат / не замужем' и 'гражданский брак' менее надежны.
  
   **3.Есть ли зависимость между уровнем дохода и возвратом кредита в срок?**
   
 
 - Видно, что люди с высоким доходом более ответственно относятся к соблюдению сроков погашения кредита.(У них проблем с            возвратом меньше). Наименее надежны клиенты со средним уровнем дохода.можно сказать, что зависимость между уровнем дохода и      возвратом кредита в срок также присутствует.Но однозначно на мой взгляд к ней не стоит относится.
  
  **4.Как разные цели кредита влияют на его возврат в срок?**
  
 
 - Выявлено,что по операциям с недвижимостью процент невозврата кредита вовремя самый низкий(**7.2%**) и число таких заемщиков      больше всего.(видимо люди, берущие кредит на эти цели более ответственны).Самый высокий процент невозврата кредита вовремя       по операциям, связанным с приобретением автомобиля и получением образования(**9.3% и 9.2% соответственно**).
  
**ИТОГ**  : Банку выгоднее и надежнее выдавать кредиты клиентам с отсутствием детей и семейным положением "женат / замужем",       со стабильным достаточно высоким доходом и на операции связанные с недвижимостью.Эти категории граждан  с большей               вероятностью смогут погасить кредиты в сроки,установленные договором.